mycore.org/content/de/documentation/basics/basics_mcr_ocfl.md

```diff
@@ -0,0 +1,312 @@
+
+Wenn alles geklappt hat, dann kann man den MetadatenManager wie folgt umstellen.
+```shell {linenos=table}
+ MCR.Metadata.Manager=org.mycore.ocfl.MCROCFLXMLMetadataManager
+```
+
+## Migration zwischen Layouts
+
+Die Migration zwischen den OCFL Layouts funktioniert genauso wie zu OCFL,
+der einzige Unterschied ist das hierbei der Metadaten-Manager des OCFL zu nutzen ist.
+
+Um eine Repository zu Migrieren, ist die Quell-Repository mit
+`MCR.Metadata.Manager.Repository={Quell-Repository}`
+zu setzen.
+
+Das Ziel Repository muss manuell gesetzt werden (siehe [Konfiguration](#konfiguration)), diese Repository darf nicht den gleichen Typ (Repository Provider) haben wie die Quelle. Wird dies trotzdem benötigt, ist es abzuwägen, ob es nicht besser ist, das Repository einfach zu kopieren.
+
+Hierbei kann es manchmal vorkommen, dass ein `MCRUsageException` ausgegeben wird. Dies tritt dann auf, wenn Objekte oder Derivate vor der Migration zu OCFL gelöscht wurden und daher nicht mit Migriert wurden. Folgend wird versucht, deren Versionsverlauf zu lesen, da MyCoRe diese ID noch immer bekannt ist. Dies tritt nicht auf wenn Objekte und Derivate nach der Migration gelöscht werden, da der Versionsverlauf nicht mit gelöscht wird.\
+Dies wird in zukünftigen Versionen behoben werden und hat <u>keinen Impakt in der Funktionsweise von OCFL</u>, und kann daher ohne Bedenken ignoriert werden.
+
+## Verfügbare Repository Layouts
+### OCFL Community Extension 0003: Hashed Truncated N-tuple Trees with Object ID Encapsulating Directory for OCFL Storage Hierarchies
+
+Das *0003-hash-and-id-n-tuple-storage-layout* ist das Standard Layout in MyCoRe. Es ist ein Layout aus den offiziellen Community Extensions für OCFL. 
+Ein Repository mit dem *0003-hash-and-id-n-tuple-storage-layout* lässt sich auch durch externe Programme einfach einsehen und verändern.\
+Aus den Dateinamen wird ein Hashwert gebildet, mit dem danach die Ordnerstruktur gebildet wird.
+
+
+Es ist zu beachten, das es vermieden werden soll, Dateien zu löschen oder neu anzulegen, da MyCoRe seine eigene Liste für die IDs führt und es daher zu Fehlern oder Datenverlust kommen kann.
+
+Der Nachteil ist, dass durch die Bildung des Hashwertes ein Dateizugriff länger dauert. Durch die Nutzung des Hashwertes ist es ohne externe Tools nicht möglich, herauszufinden, in welchem Ordner sich welches Objekt befindet, sondern nur durch den root Ordner des Objektes und der `inventory.json` .
+
+Link zur Spezifikation: [0003-hash-and-id-n-tuple-storage-layout.md <sup><i class="fas fa-external-link-alt"></i></sup>](https://github.com/OCFL/extensions/blob/main/docs/0003-hash-and-id-n-tuple-storage-layout.md)
+
+#### Hash Generation
+
+Die Hashes im Implementierten *0003-hash-and-id-n-tuple-storage-layout* werden mit Sha-256 Generiert, 3 Tuple aus jeweils 3 Stellen generieren dann den Pfad zum Objekt.\
```

Comment:
Der Hashwert im *0003-hash-and-id-n-tuple-storage-layout* wird mit mit dem Algorithmus SHA-256 generiert. 3 Tupel mit jeweils 3 Stellen vom Beginn des Hashwertes bilden dann den Pfad zum Objekt.

Der Hashwert wird aus dem ursprünglichen Dateinamen gebildet. Anschließend wird ein sicherer Dateiname erzeugt, indem Sonderzeichen durch ihre Escape-Sequenzen codiert werden. Dadurch wird eine problemlose Speicherung gewährleistet.